### PR TITLE
CLC-5516 Maintain admins on bCourses test systems

### PR DESCRIPTION
**Pull request by raydavis**
https://jira.ets.berkeley.edu/jira/browse/CLC-5516

- New shell script to centralize reconfiguration of overwritten Canvas servers
- Let any Canvas proxy point to a non-default server
- Remove obsolete shell scripts

### CODE REVIEW

File: app/models/canvas/admins.rb [hunk 1]
```diff
@@ -8,23 +21,53 @@ def initialize(options = {})
...
-    def admin_user?(uid)
-      admins = self.class.fetch_from_cache(@account_id) { paged_get request_path }
+    def admin_user?(uid, cached = true)
+      if cached
+        admins = self.class.fetch_from_cache(@account_id) { admins_list }
+      else
+        # When editing remote account admin lists, we must access the most recent data and not cache it locally.
+        admins = admins_list
+      end
       admins[:body].present? && admins[:body].index {|acct| acct['user']['sis_login_id'] == uid.to_s}.present?
```
Comment (paulkerschen) [inline app/models/canvas/admins.rb]: Canvas::Proxy now includes an `optional_cache` method (usage example in Canvas::Course) that can condense this logic a bit.

**Comment by paulkerschen** [comment]
@raydavis I did note the newish optional caching idiom above, but as a whole this PR looks highly sensible to me.

**Comment by raydavis** [comment]
Thank you for the optional_cache pointer, @paulkerschen. PR amended accordingly.

File: app/models/canvas/admins.rb [hunk 2]
```diff
@@ -8,23 +21,53 @@ def initialize(options = {})
...
+      if admin_user? canvas_login_id, false
+        added = false
+      else
+        profile = Canvas::SisUserProfile.new(user_id: canvas_login_id).get
+        canvas_user_id = profile['id']
+        add_admin canvas_user_id
+        added = true
+      end
+      {added: added}
+    end
```
Comment (paulkerschen) [inline app/models/canvas/admins.rb]: As discussed in person, we see no need to get fussy about placing this bit of business logic in a separate class from the proxy call proper.

**Pull request merged**